crates/miden-objects/src/batch/batch_id.rs

```diff
@@ -0,0 +1,65 @@
+use alloc::{string::String, vec::Vec};
+
+use vm_core::{Felt, ZERO};
+use vm_processor::Digest;
+
+use crate::{
+    account::AccountId,
+    transaction::{ProvenTransaction, TransactionId},
+    Hasher,
+};
+
+// BATCH ID
+// ================================================================================================
+
+/// Uniquely identifies a batch of transactions, i.e. both
+/// [`ProposedBatch`](crate::batch::ProposedBatch) and [`ProvenBatch`](crate::batch::ProvenBatch).
+///
+/// This is a sequential hash of the tuple `(TRANSACTION_ID || [account_id_prefix,
+/// account_id_suffix, 0, 0])` of all transactions and the accounts their executed against in the
+/// batch.
+#[derive(Debug, Copy, Clone, Eq, Ord, PartialEq, PartialOrd)]
+pub struct BatchId(Digest);
+
+impl BatchId {
+    /// Calculates a batch ID from the given set of transactions.
+    pub fn compute_from_transactions<'tx, T>(txs: T) -> Self
+    where
+        T: Iterator<Item = &'tx ProvenTransaction>,
+    {
+        Self::compute_from_ids(txs.map(|tx| (tx.id(), tx.account_id())))
+    }
+
+    /// Calculates a batch ID from the given transaction ID and account ID tuple.
+    pub fn compute_from_ids(iter: impl Iterator<Item = (TransactionId, AccountId)>) -> Self {
```

Comment:
nit: do we need the `compute_` prefix for these? Maybe just `from_transactions()` and `from_ids()` would be fine?

Also, as mentioned in the `miden-node` PR review, maybe we should have a wrapper struct for `(TransactionId, AccountId)` tuple (if we can come up with a good name for it).

Reply:
Sounds good, removed the compute prefix.

It would be nice to have such a struct, but I can't come up with a good name. It's a very specific combination.
The straightforward name would be `TxAccountPair`, but it's just a name to have a name.

Reply:
> It would be nice to have such a struct, but I can't come up with a good name. It's a very specific combination.
> The straightforward name would be `TxAccountPair`, but it's just a name to have a name.

Yeah, I can't come up with anything good either. Let's keep it as is for now.